identity-service/src/notifications/processNotifications/reactionNotification.js

```diff
@@ -0,0 +1,50 @@
+const models = require('../../models')
+const { notificationTypes } = require('../constants')
+
+async function processReactionNotifications (notifications, tx) {
+  for (const notification of notifications) {
+    const { slot, initiator: reactorId, metadata: { reaction_value: reactionValue, reacted_to: reactedTo, reaction_type: reactionType } } = notification
+
+    // First find the originating tip receive notif
+    const receiveTipNotification = await models.SolanaNotification.findOne({
+      where: {
+        type: notificationTypes.TipReceive,
+        metadata: {
+          [models.Sequelize.Op.contains]: {
+            'tipTxSignature': reactedTo
+          }
+        }
+      }
+    })
+
+    if (!receiveTipNotification) {
+      console.error(`Received reaction to ${reactedTo} from ${reactorId}, but not corresponding tip notification`)
+      return notifications
+    }
+
+    // Set it's reaction value
+    receiveTipNotification.metadata = { ...receiveTipNotification.metadata, reactionValue }
```

Comment:
just wondering about doing `receiveTipNotification.metadata.reactionValue = reactionValue`

Reply:
yes, you would think! this is what I initially did. for some reason sequelize didn't pick up that the model had changed, and calling .save() didn't do anything. took me forever to figure this out. I'll add a comment explaining